app/controllers/sessions_controller.rb

```diff
@@ -67,6 +69,21 @@ def authenticate
     redirect_to root_path if signed_in?
   end
 
+  def trusted_launch
+    handle_with(TrustedParametersLaunch,
+                user_state: self,
```

Comment:
Pass in `trusted_state: session[:trusted]` here so that don't have to say `@attrs = request.session['trusted']` in the handler below.  Makes testing easier too -- don't have to create or mock an entire request.